nuggetsapp/models.py

```diff
@@ -59,5 +66,21 @@ def add_new_nugget_for_user(cls, nugget, user, is_owner=False):
 
     @classmethod
     def get_nuggets_by_user(cls, user):
-        return [x.nugget for x in cls.objects.filter(user = user)]
+        return [x.nugget for x in cls.objects.filter(user = user).filter(is_deleted = False)]
```

Comment:
I think I prefer filter(Q(A) & Q(B) | Q(C)) because it's more explicit and supports both and/or.